### ISSUE TITLE
float(SVector) gives an MVector

### ISSUE THREAD

**Issue by stevengj** [JuliaArrays/StaticArrays.jl#247]
```
julia> float(SVector(0))
1-element MVector{1,Float64}:
 0.0
```
This is because it is calling `base/float.jl`, which calls `convert(AbstractArray, ...)` which calls `similar`, I think(?), but in this case it seems more useful to return an `SVector`?

Or is there a better way to promote an `SVector` to a float-based type, e.g. from `Complex{Int}` to `Complex{Float}`?

**Comment by c42f** [comment]
How about

```
float.(SVector(0))
```

I guess we should just add the following as a general solution:

```
float(a::StaticArray) = map(float,a)
```

Perhaps this would also be a better solution for Base for AbstractArrays of non-floats.

**Comment by daanhb** [comment]
I think this one is fixed, perhaps by #747?
```jl
julia> float(SVector(0))
1-element SArray{Tuple{1},Float64,1,1} with indices SOneTo(1):
 0.0
```
